src/app/vertextool/qgsvertextool.cpp

```diff
@@ -533,7 +533,21 @@ void QgsVertexTool::cadCanvasReleaseEvent( QgsMapMouseEvent *e )
       if ( mLockedFeature && mLockedFeature->layer() != vlayer )
         continue;  // with locked feature we only allow selection of its vertices
 
-      QgsRectangle layerRect = toLayerCoordinates( vlayer, rubberBandGeometry.boundingBox() );
+      QgsGeometry layerRubberBandGeometry = rubberBandGeometry;
+      try
+      {
+        QgsCoordinateTransform ct = mCanvas->mapSettings().layerTransform( vlayer );
+        if ( ct.isValid() )
+          layerRubberBandGeometry.transform( ct, QgsCoordinateTransform::ReverseTransform );
+      }
+      catch ( QgsCsException & )
+      {
+      }
```

Comment:
Does it make sense to continue with the untransformed geometry if an exception happens?

Reply:
True -- fixed.